### ISSUE TITLE
Extract Accuracy/Radius

### ISSUE THREAD

**Issue by mrx23dot** [darkosancanin/apple_bssid_locator#3]
Does anyone know which field encodes Radius?

```
    _descriptor.FieldDescriptor(
      name='latitude', full_name='WifiDevice.Location.latitude', index=0,
      number=1, type=3, cpp_type=2, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      options=None),
    _descriptor.FieldDescriptor(
      name='longitude', full_name='WifiDevice.Location.longitude', index=1,
      number=2, type=3, cpp_type=2, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      options=None),
    _descriptor.FieldDescriptor(
      name='unknown_value3', full_name='WifiDevice.Location.unknown_value3', index=2,
      number=3, type=3, cpp_type=2, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      options=None),
    _descriptor.FieldDescriptor(
      name='unknown_value4', full_name='WifiDevice.Location.unknown_value4', index=3,
      number=4, type=3, cpp_type=2, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      options=None),
    _descriptor.FieldDescriptor(
      name='unknown_value5', full_name='WifiDevice.Location.unknown_value5', index=4,
      number=5, type=3, cpp_type=2, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      options=None),
    _descriptor.FieldDescriptor(
      name='unknown_value6', full_name='WifiDevice.Location.unknown_value6', index=5,
      number=6, type=3, cpp_type=2, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      options=None),
    _descriptor.FieldDescriptor(
      name='unknown_value7', full_name='WifiDevice.Location.unknown_value7', index=6,
      number=7, type=3, cpp_type=2, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      options=None),
    _descriptor.FieldDescriptor(
      name='unknown_value8', full_name='WifiDevice.Location.unknown_value8', index=7,
      number=8, type=3, cpp_type=2, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      options=None),
    _descriptor.FieldDescriptor(
      name='unknown_value9', full_name='WifiDevice.Location.unknown_value9', index=8,
      number=9, type=3, cpp_type=2, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      options=None),
    _descriptor.FieldDescriptor(
      name='unknown_value10', full_name='WifiDevice.Location.unknown_value10', index=9,
      number=10, type=3, cpp_type=2, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      options=None),
    _descriptor.FieldDescriptor(
      name='unknown_value11', full_name='WifiDevice.Location.unknown_value11', index=10,
      number=11, type=3, cpp_type=2, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      options=None),
    _descriptor.FieldDescriptor(
      name='unknown_value12', full_name='WifiDevice.Location.unknown_value12', index=11,
      number=12, type=3, cpp_type=2, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      options=None),
    _descriptor.FieldDescriptor(
      name='unknown_value21', full_name='WifiDevice.Location.unknown_value21', index=12,
      number=21, type=3, cpp_type=2, label=1,
      has_default_value=False, default_value=0,
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      options=None),
```

**Comment by biemster** [comment]
@mrx23dot did you figure this one out? I only recently found out about this when some paper got quite a lot of attention, but this would nicely combine with the FindMy stuff I would say.

**Comment by biemster** [comment]
i was thinking about tags just broadcasting a wifi bssid instead / in addition to a ble advertisement..

would require a different radio but many mcu's nowadays do wifi as well (with esp32 the best known example)

**Comment by biemster** [comment]
The tag would just pretend to be a router by sending out a bssid, then the script from this repo can be used to get it's location based on the mac address it uses for that. Am I missing something here?

**Comment by acheong08** [comment]
> The tag would just pretend to be a router by sending out a bssid, then the script from this repo can be used to get it's location based on the mac address it uses for that. Am I missing something here?

That's what I'm currently attempting. It seems that it isn't immediately reported. I've been MITMing my iPhone for a day or two now and haven't caught any requests with the GPS location of a hotspot I've set up (and requests for that BSSID isn't returning anything). It might take a while for it to gain confidence in the location which makes sense considering `confidence` shows up a ton in the `CoreLocation`/`locationd` binaries.

**Comment by acheong08** [comment]
> Weirdly apple only allows one bssid to be looked up. (cannot triangulate)

What do you mean? You can send multiple BSSIDs at once & Apple returns the GPS locations for all of them. There is also an option in the protobuf which makes Apple return all BSSIDs and their locations within the "neighborhood group" (I get ~200 from my BSSID) which allows you to triangulate based on signal strength

**Comment by acheong08** [comment]
In Golang: https://github.com/acheong08/apple-corelocation-experiments/blob/main/lib/appledb.go

**Comment by acheong08** [comment]
I've trying all sorts of things to register a BSSID onto their database to no avail. My iPhone with all analytics turned on isn't reporting my hotspot at all with MAC spoofed to a valid vendor.

Did find something interesting though: https://gspe85-ssl.ls.apple.com/wifi_request_tile returns a bunch of GPS locations and BSSIDs (encoded in int64). The request includes a tile ID which means you can literally look at what BSSIDs are in any given area (currently trying to figure out the logic behind tileIDs)

**Comment by acheong08** [comment]
> wigle.net

Oh I'm not trying to do anything practical. Reverse engineering is just fun for me. Also, wigle has terrible rate limits, sell your data for commercial purposes, and doesn't provide data to researchers. I have a slight distaste for them

> Also try to spoof a home router, maybe one that's already in the system and see if it start moving.

Yup doing that now. Waiting for something to happen.

I'm also trying to brute force the API to see which fields might allow me to upload data. Currently most of them just returns 400s or default missing values.

**Comment by acheong08** [comment]
> The design of Apple's system allowed Rye and Levin to compile a database of 490 million BSSIDs around the world, which they could then use to track the movements of individuals and groups of people over time.

I actually found an API that allows me to request BSSIDs for any arbitrary GPS location. Just finished figuring out the decoding process for `x-tilekey` in `GEOTileKeyMake`. I've documented everything [here](https://github.com/acheong08/apple-corelocation-experiments?tab=readme-ov-file#wifi-tile) including an interactive demo which allows you to click anything on a map and it'll return the closest cluster of BSSIDs.

Will try to reproduce their experiment more efficiently and write something up. Will get back to reverse engineering the process for how BSSIDs are picked up next.
